src/pykeen/triples/triples_numeric_literals_factory.py

```diff
@@ -45,60 +46,73 @@ def create_matrix_of_literals(
 class TriplesNumericLiteralsFactory(TriplesFactory):
     """Create multi-modal instances given the path to triples."""
 
+    file_name_literal_to_id: ClassVar[str] = "literal_to_id"
+    file_name_numeric_literals: ClassVar[str] = "literals"
+
     def __init__(
         self,
         *,
-        path: Union[None, str, pathlib.Path, TextIO] = None,
-        triples: Optional[LabeledTriples] = None,
-        path_to_numeric_triples: Union[None, str, pathlib.Path, TextIO] = None,
-        numeric_triples: Optional[np.ndarray] = None,
+        numeric_literals: np.ndarray,
+        literals_to_id: Mapping[str, int],
         **kwargs,
     ) -> None:
         """Initialize the multi-modal triples factory.
 
-        :param path: The path to a 3-column TSV file with triples in it. If not specified,
-         you should specify ``triples``.
-        :param triples:  A 3-column numpy array with triples in it. If not specified,
-         you should specify ``path``
-        :param path_to_numeric_triples: The path to a 3-column TSV file with triples and
-         numeric. If not specified, you should specify ``numeric_triples``.
-        :param numeric_triples:  A 3-column numpy array with numeric triples in it. If not
-         specified, you should specify ``path_to_numeric_triples``.
+        :param numeric_literals: shape: (num_entities, num_literals)
+            the numeric literals as a dense matrix.
+        :param literals_to_id:
+            a mapping from literal names to their IDs, i.e., the columns in the `numeric_literals` matrix.
+        :param kwargs:
+            additional keyword-based parameters passed to :meth:`TriplesFactory.__init__`.
         """
-        if path is not None:
-            base = TriplesFactory.from_path(path=path, **kwargs)
-        elif triples is None:
-            base = TriplesFactory(**kwargs)
-        else:
-            base = TriplesFactory.from_labeled_triples(triples=triples, **kwargs)
-        super().__init__(
+        super().__init__(**kwargs)
+        self.numeric_literals = numeric_literals
+        self.literals_to_id = literals_to_id
+
+    @classmethod
+    def from_path(
+        cls,
+        path: Union[str, pathlib.Path, TextIO],
+        *,
+        path_to_numeric_triples: Union[str, pathlib.Path, TextIO] = None,
+        **kwargs,
+    ) -> "TriplesNumericLiteralsFactory":  # noqa: D102
+        if path_to_numeric_triples is None:
+            raise ValueError(f"{cls.__name__} requires path_to_numeric_triples.")
+        numeric_triples = load_triples(path_to_numeric_triples)
+        triples = load_triples(path)
+        return cls.from_labeled_triples(triples=triples, numeric_triples=numeric_triples, **kwargs)
+
+    @classmethod
+    def from_labeled_triples(
+        cls,
+        *,
+        triples: LabeledTriples,
+        numeric_triples: LabeledTriples = None,
```

Comment:
Is there a meaningful case in which `numeric_triples` could be none?

Reply:
> Is there a meaningful case in which numeric_triples could be none?

no

>Is this so it can follow the parent's signature?

yes